src/HalResource.php

```diff
@@ -412,7 +390,7 @@ private function firstResource(array $resources): ?HalResource
     /**
      * @param null|object|HalResource[] $value
      */
-    private function isResourceCollection($value, string $name, string $context): bool
+    private function isResourceCollection($value): bool
```

Comment:
Method could be  simplified by returning from array_reduce directly

Reply:
8eb7dffb6cc6d6327e1e149309c5f81120be30dd